### ISSUE TITLE
Restarting agents from master node doesn't work

### ISSUE THREAD

**Issue by mark260486** [wazuh/wazuh#5245]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 3.12.2 | ossec-remoted | Manager | Packages | Amazon Linux 2 |

Hi team!
We were found the following issue:

We can restart agent with the API request, but cannot with ```/var/ossec/bin/agent_control -R -u $agent_id```
API request:
```
curl -u foo:bar -k -X PUT -d '{"command":"restart-ossec0", "arguments": ["-", "null", "(from_the_server)", "(no_rule_id)"]}' -H 'Content-Type:application/json' "https://127.0.0.1:55000/active-response/$agent_id?pretty"
```
In ossec.log we have the following errors both for Linux and Windows agents:
```
ossec-remoted[2744] sendmsg.c:112 at send_msg(): DEBUG: Send operation cancelled due to closed socket.
ossec-remoted[2744] sendmsg.c:89 at send_msg(): WARNING: (1245): Sending message to disconnected agent '1666'.
```

All this happen in case if agent is connected to the Worker node. If agent connected to the Master node, all is good.

We will very appreciate some help. Thank you.

**Comment by palaciosjeremias** [comment]
Hi mark260486, thanks for your feedback,
We have tested the described scenario and were able to reproduce this behavior.
This happens because on clustered environments, **Agents** communicates to the configured worker **remoted** daemon. But **agent_control** is trying to send the restart request thru **remoted** on master node, which doesn't have an open communication with **Agent**.
We will investigate this further and bring a response ASAP.
Regards.

**Comment by davidjiglesias** [comment]
Hello @mark260486,

Adding to what @palaciosjeremias said, I would suggest you always try to use the API to perform restart operations in a clustered mode.

Apart from the example you mentioned using active-response, the best use to restart agents are restart endpoints. You can use any of the following depending on your needs:
- [Restart a list of agents](https://documentation.wazuh.com/3.12/user-manual/api/reference.html#restart-a-list-of-agents)
- [Restart all agents belonging to a group](https://documentation.wazuh.com/3.12/user-manual/api/reference.html#restart-agents-which-belong-to-a-group)
- [Restart all agents](https://documentation.wazuh.com/3.12/user-manual/api/reference.html#restart-a-list-of-agents)

These API endpoints are fully developed and tested in a clustered environment using distributed API calls. We recommend always using API endpoints over binaries when available, as the API is intended to fully deprecate binaries at some point in the future.

I hope you find this information useful. If you have any other questions please do not hesitate to ask.

Best regards,

David J. Iglesias

**Comment by mark260486** [comment]
Hello, @palaciosjeremias and @davidjiglesias! Thank you for the reply and advices. We will await further for your answers.

Best regards, Michael

**Comment by palaciosjeremias** [comment]
Hi @mark260486,
As @davidjiglesias detailed above, Wazuh API is intended to fully deprecate binaries, so we enhance you to use Wazuh API to control **Agents** restarts.
Based on you observation, we decided to improve our documentation to clarify this limitation of the binaries in clustered configurations:
https://github.com/wazuh/wazuh-documentation/issues/2591

Please, let us know if you have further questions.
Regards.

**Comment by mark260486** [comment]
Hi @palaciosjeremias, thank you for the reply!

I think that in this case we need to plan the transition to API, since we have some automation with the help of binaries. We hope that you can fix the agent_control issue and we will monitor further updates of Wazuh. If you think it is necessary, I think that the issue can be closed. Thanks for the help,

regards, Michael.

**Comment by Abykaam** [comment]
Is this issue is fixed in the wazuh agent 4.3.10 and above or not
